core/app/models/spree/promotion_action.rb

```diff
@@ -19,5 +19,15 @@ class PromotionAction < Spree::Base
     def perform(options = {})
       raise 'perform should be implemented in a sub-class of PromotionAction'
     end
+
+    def duplicate
+      duplicated_action = self.dup
+      duplicate_calculator(duplicated_action) if try(:calculator)
+      duplicated_action
+    end
+
+    def duplicate_calculator(dupe)
```

Comment:
Agree about names, I guess my thought here is that if a subclass decided to do something different with the calculator, it would be easier to implement your own duplicate_calculator instead of rewriting the whole method.